internals/src/macros.rs

```diff
@@ -139,9 +139,15 @@ macro_rules! debug_from_display {
 /// Asserts a boolean expression at compile time.
 #[macro_export]
 macro_rules! const_assert {
-    ($x:expr) => {
-        const _: [(); 0 - !$x as usize] = [];
-    };
+    ($x:expr $(; $message:expr)?) => {
+        const _: bool = {
```

Comment:
Good to call this out so that if we later dig into this, we at least can see that we noticed it and thought it was funny at the time.

But it really makes no difference at all. I suspect it could have also been `!` if you want the simplest possible type.

Reply:
It'd have to always panic to be `!`.

Reply:
Seemed logical to me that if one is asserting an expression then the expression is a bool. I agree with the observation that it is not technically required.

Reply:
FTR none of the asserts in `std` evaluates to `bool` for a simple reason: the return value would be always `true`. Same goes here except you can't even compile a `const` that's not `true`, that's why it carries no information which is better represented by `()` type.